src/parquet/arrow/reader.cc

```diff
@@ -562,6 +566,31 @@ Status FileReader::Impl::ReadTable(std::shared_ptr<Table>* table) {
   return ReadTable(indices, table);
 }
 
+Status FileReader::Impl::ReadRowGroups(const std::vector<int>& row_groups,
+                                       const std::vector<int>& indices,
+                                       std::shared_ptr<Table>* table) {
+  // TODO(PARQUET-1393): Modify the record readers to already read this into a single,
+  // continuous array.
+  std::vector<std::shared_ptr<Table>> tables;
+
+  for (size_t i = 0; i < row_groups.size(); ++i) {
+    std::shared_ptr<Table> rg_table;
+    RETURN_NOT_OK(ReadRowGroup(row_groups[i], indices, &rg_table));
```

Comment:
a small optimization here is to use
`std::vector<std::shared_ptr<Table>> tables(row_group.size(), nullptr)` above and
use `&tables[i]` here. Afaik certain tools like Spark generate files with many small RowGroups.

Reply:
I just saw your TODO. So this change might be less relevant.

Reply:
Still, added it as it saves some cycles.